pkgs/development/tools/protoc-gen-rust/default.nix

```diff
@@ -0,0 +1,37 @@
+{ fetchFromGitHub
+, lib
+, rustPlatform
+, protobuf
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "protoc-gen-rust";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "stepancheg";
+    repo = "rust-protobuf";
+    sha256 = "sha256-oMcemWuhj9ZRAt93qEkgVghpNaC7TvRDHA9ExthS7Is=";
+    rev = "v${version}";
+  };
```

Comment:
We can use `fetchCrate` here to avoid adding our own `Cargo.lock` -- you'll want to fetch the [protobuf-codegen](https://lib.rs/crates/protobuf-codegen) crate.

Reply:
Done! Agreed that this is a better approach.